File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CompletionBookkeepingEvent.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class CompletionBookkeepingEvent(
  val id: CompletionLogID,
  val startedAt: Long,
  val networkRequestStartedAt: Long? = null,
  val startLoggedAt: Long? = null,
  val loadedAt: Long? = null,
  val suggestedAt: Long? = null,
  val suggestionLoggedAt: Long? = null,
  val suggestionAnalyticsLoggedAt: Long? = null,
  val acceptedAt: Long? = null,
  val items: List<CompletionItemInfo>,
  val loggedPartialAcceptedLength: Long,
  val read: Boolean,
)

File: agent/bindings/csharp/CompletionBookkeepingEvent.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class CompletionBookkeepingEvent
  {
    [JsonProperty(PropertyName = "id")]
    public CompletionLogID Id { get; set; }
    [JsonProperty(PropertyName = "startedAt")]
    public int StartedAt { get; set; }
    [JsonProperty(PropertyName = "networkRequestStartedAt")]
    public int NetworkRequestStartedAt { get; set; }
    [JsonProperty(PropertyName = "startLoggedAt")]
    public int StartLoggedAt { get; set; }
    [JsonProperty(PropertyName = "loadedAt")]
    public int LoadedAt { get; set; }
    [JsonProperty(PropertyName = "suggestedAt")]
    public int SuggestedAt { get; set; }
    [JsonProperty(PropertyName = "suggestionLoggedAt")]
    public int SuggestionLoggedAt { get; set; }
    [JsonProperty(PropertyName = "suggestionAnalyticsLoggedAt")]
    public int SuggestionAnalyticsLoggedAt { get; set; }
    [JsonProperty(PropertyName = "acceptedAt")]
    public int AcceptedAt { get; set; }
    [JsonProperty(PropertyName = "items")]
    public CompletionItemInfo[] Items { get; set; }
    [JsonProperty(PropertyName = "loggedPartialAcceptedLength")]
    public int LoggedPartialAcceptedLength { get; set; }
    [JsonProperty(PropertyName = "read")]
    public bool Read { get; set; }
  }
}
